src/richie-front/js/utils/filters/getActiveFilterValues.ts

```diff
@@ -10,22 +15,18 @@ export const getActiveFilterValues = (
 ): FilterValue[] => {
   let currentValues = currentParams[machineName];
 
+  if (!currentValues) {
+    return [];
+  }
+
+  // Wrap base typed values in arrays to make manipulating them easier
+  if (typeof currentValues === 'string' || typeof currentValues === 'number') {
+    currentValues = [currentValues];
+  }
+
   switch (machineName) {
```

Comment:
Now that we have implemented both sides, I think we could make this switch more generic with something like

```
switch (true) {
  case includes(Object.keys(FILTERS_RESOURCES), machineName):
    // ...

  case includes(Object.keys(FILTERS_HARDCODED), machineName):
    // ...
}
```

You did not change this code in your PR but I think it's in scope to stop hardcoding filter names here and use names from settings instead 😇

Reply:
I tried to change it and it works.

But I have some trouble with typescript because it doesn't know anymore what can be the value of `machineName` and I have this error :

```
 Element implicitly has an 'any' type because type '{ courses?: Maybe<ResourceByIdState<Course> & ResourceListState<Course>>; organizations?: Maybe<ResourceByIdState<Organization> & ResourceListState<Organization>>; subjects?: Maybe<...>; }' has no index signature.
```

I looked on google but I see nothing allowing me to tell to typescript that here `machineName` is type `resourceBasedFilterGroupName` and in the other case of type `hardcodedFilterGroupName`. Do you know if there is an annotation or something else to do that ?

Thanks

Reply:
Indeed it does not seem that typescript is able to either interpret `includes(Object.keys(FILTERS_RESOURCES), machineName)` as type discrimination, nor to apply type discrimination to an inverted switch like this.

What we can do is use a type guard and replace the switch with an `if/else`. Since we already need it in two separate places the type guard should probably be in `utils`.

Since the scope on this is increasing a bit it could probably be done in a separate PR though, so if you just fix the remaining unrelated comment this we should be ready to merge 👍

Reply:
ok, I just push a commit resolving the other comments.